packages/vscode-dependencies-validation/src/autofix/depsIssues.ts

```diff
@@ -0,0 +1,59 @@
+import type { Uri } from "vscode";
+import {
+  VscodeFileEventConfig,
+  VscodeUriFile,
+  VscodeWorkspace,
+} from "../vscodeTypes";
+import {
+  ENABLE_AUTOFIX,
+  getAutoFixDelay,
+  isAutoFixEnabled,
+} from "./configuration";
+import { addPackageJsonFileWatcher as addPackageJsonFileWatcher } from "./packageJsonFileWatcher";
+import { addUnsupportedFilesWatcher } from "./unsupportedFilesWatcher";
+import { findAndFixDepsIssues } from "../util";
+import { clearDiagnostics } from "../util";
+
+export function activateDepsIssuesAutoFix(
+  vscodeConfig: VscodeFileEventConfig & VscodeUriFile
+): void {
+  fixWorkspaceDepsIssues(vscodeConfig);
+  addPackageJsonFileWatcher(vscodeConfig);
+  addUnsupportedFilesWatcher(vscodeConfig);
```

Comment:
My understanding is that `addUnsupportedFilesWatcher` is meant to handle the edge case
of a package.json file which stopped being **exlcuded** due to the existance of none npm files (e.g yarn.lock).

My Feedback:
- Long term watcher logic will be insufficent because we may have more complex conditions
  - on when/what is being excluded.
- evaluate removing this "unsupported' watcher logic completely and **replacing** it with periodic checks (e.g hourly). reasons:
   - Reduce solution complexity.
   - Would still be correct, just delayed.
   - Avoids future problem of expanding advanced detection of unsupported flows.
   - Periodic checks will also detect edge cases we have not considered.
   
One problem with periodic flows is that it could cause race conditions with user operations.
- e.g a periodic check happening just when an end user wrote `rm -rf node_modules` but before they did `npm install` again.

Reply:
I do not think we need periodic checks here. 
Let's talk about it